charts/esphome/values.yaml

```diff
@@ -0,0 +1,154 @@
+# Default values for esphome.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
+image:
+  repository: esphome/esphome
+  tag: 1.14.5
+  pullPolicy: IfNotPresent
+  pullSecrets: []
+
+# upgrade strategy type (e.g. Recreate or RollingUpdate)
+strategyType: Recreate
+
+# Probes configuration
+probes:
+  liveness:
+    enabled: true
+    scheme: HTTP
+    initialDelaySeconds: 60
+    failureThreshold: 5
+    timeoutSeconds: 10
+  readiness:
+    enabled: true
+    scheme: HTTP
+    initialDelaySeconds: 60
+    failureThreshold: 5
+    timeoutSeconds: 10
+  startup:
+    enabled: false
+    scheme: HTTP
+    failureThreshold: 30
+    periodSeconds: 10
+service:
+  type: ClusterIP
+  port: 6052
+  portName: api
+  additionalPorts: []
+  # - name: homematicproxy
+  #   port: 2001
+  #   targetPort: 2001
+  annotations: {}
+  labels: {}
+  clusterIP: ""
+  ## List of IP addresses at which the hass-configurator service is available
+  ## Ref: https://kubernetes.io/docs/user-guide/services/#external-ips
+  ##
+  externalIPs: []
+  loadBalancerIP: ""
+  loadBalancerSourceRanges: []
+  # nodePort: 30000
+  publishNotReadyAddresses: false
+
+ingress:
+  enabled: false
+  annotations: {}
+    # kubernetes.io/ingress.class: nginx
+    # kubernetes.io/tls-acme: "true"
+  path: /
+  hosts:
+    - esphome.local
+  tls: []
+  #  - secretName: esphome-tls
+  #    hosts:
+  #      - esphome.local
+
+hostNetwork: false
+
+persistence:
+  enabled: true
+  ## esphome data Persistent Volume Storage Class
+  ## If defined, storageClassName: <storageClass>
+  ## If set to "-", storageClassName: "", which disables dynamic provisioning
+  ## If undefined (the default) or set to null, no storageClassName spec is
+  ##   set, choosing the default provisioner.  (gp2 on AWS, standard on
+  ##   GKE, AWS & OpenStack)
+  ##
+  # storageClass: "-"
+  ##
+  ## If you want to reuse an existing claim, you can pass the name of the PVC using
+  ## the existingClaim variable
+  # existingClaim: your-claim
+  ##
+  ## If you want to use a volume on the host machine instead of a PVC:
+  # hostPath: /path/to/the/config/folder
+  accessMode: ReadWriteOnce
+  size: 5Gi
```

Comment:
If the configuration data is relatively small, we may want to set the default size here to something smaller (maybe `1Gi` ?)

Reply:
The config directory also includes compiled data by default. There's a way to move that data to a different location on a per-node basis but I'm not sure if that can be done at a global level. At any rate, you're correct that it should be a small volume.